### ISSUE TITLE
[BUG] Exception when using to_numeric with Boolean mask

### ISSUE THREAD

**Issue by mlahir1** [rapidsai/cudf#11842]
Repro code:
```
import cudf
df = cudf.DataFrame()
df['item_id'] = ['0101', '102','103','10a4','|105','106']
df['upc'] = ['123', '208001','20002','25103','201004','200005']
df = df[~df.item_id.str.startswith('0')]
df['item_id'] = cudf.to_numeric(df.item_id, errors='coerce')
df
```

Exception:
```
File /conda/envs/rapids-env/lib/python3.9/site-packages/cudf/utils/dtypes.py:247, in to_cudf_compatible_scalar(val, dtype)
    244     return val
    246 if not cudf.api.types._is_scalar_or_zero_d_array(val):
--> 247     raise ValueError(
    248         f"Cannot convert value of type {type(val).__name__} "
    249         "to cudf scalar"
    250     )
    252 if isinstance(val, Decimal):
    253     return val

ValueError: Cannot convert value of type NotImplementedType to cudf scalar
````

cudf version : '22.10.00a+241.g0528b38f2b'

If i remove the boolean Mask everything runs fine:
```
import cudf
df = cudf.DataFrame()
df['item_id'] = ['0101', '102','103','10a4','|105','106']
df['upc'] = ['123', '208001','20002','25103','201004','200005']
# df = df[~df.item_id.str.startswith('0')]
df['item_id'] = cudf.to_numeric(df.item_id, errors='coerce')
df

```

Result:
```
  | item_id | upc
-- | -- | --
101.0 | 123
102.0 | 208001
103.0 | 20002
<NA> | 25103
<NA> | 201004
106.0 | 200005
```

**Comment by galipremsagar** [comment]
Hi @mlahir1,

If you are doing a source build please pull in the latest changes which contains a fix(https://github.com/rapidsai/cudf/pull/11816) to this issue. 
If you are using nightly packages, could you recreate your environment with new packages as this has been already fixed in `22.10` and a patch release has been made to `22.08.01`.

**Comment by mlahir1** [comment]
that latest 22.10 resolve the issue @galipremsagar
